test/code/plugins/VMInsightsDataCollector_subtest.rb

```diff
@@ -315,16 +315,19 @@ def test_get_filesystems_garbage
                 DfGarbage.new("/dev/foo#{__LINE__} ext4 42 42 17 3 shazam"),
                 DfGarbage.new("/dev/foo#{__LINE__} ext4 not_int 42 17 3 /shazam"),
                 DfGarbage.new("/dev/foo#{__LINE__} ext4 42 42 not_int 3 /shazam"),
+                DfGarbage.new("/dev/foo#{__LINE__} xfs 42 42 not_int 3 /shazam"),
                 DfGarbage.new("/dev/foo#{__LINE__} ext4 0x42 42 17 4 /shazam"),
                 DfGarbage.new("/dev/foo#{__LINE__} ext4 42 42 0x17 4 /shazam"),
                 DfGarbage.new("/dev/foo#{__LINE__} text4 42 42 17 4 /shazam"),
                 DfGarbage.new("/dev/foo#{__LINE__} ext4b 42 42 17 4 /shazam"),
             ].concat(expected)
             # these look funky, but they should be interpreted as decimal numbers
             stuff << DfGarbage.new("/dev/bar1 ext4 042 42 17 3 /shazam")
-            expected << Fs.new("/dev/bar1", "/shazam", 42, 17, 4)
+            expected << Fs.new("/dev/bar1", "/shazam", 42, 17, "ext4")
             stuff << DfGarbage.new("/dev/bar2 ext4 42 42 017 2 /shazam")
-            expected << Fs.new("/dev/bar2", "/shazam", 42, 17, 4)
+            expected << Fs.new("/dev/bar2", "/shazam", 42, 17, "ext4")
+            stuff << DfGarbage.new("/dev/bar3 xfs 42 42 017 2 /shazam")
+            expected << Fs.new("/dev/bar3", "/shazam", 42, 17, "xfs")
```

Comment:
Our original test case is broken. We should not be creating an invalid scenario of two devices on the same mount point.